### PR TITLE
distribute current time correctly to agents running in shadow container in different process

### PR DESCRIPTION
**Pull request by maurerle**
While using same_process=False works on current main, this PR fixes the distribution of the current time to the other container.

Before, the output agent did not block when joining the dataframes but did not execute its schedule at all (because he did not receive updates of the current time)

### CODE REVIEW

**Review by adamsjohanna — COMMENTED** [review]
I tried running the small example but run into an error.. Does it run for you? Or is there anything I need to update before running?

**Comment by maurerle** [comment]
Well, I do not have a Windows development machine at hand..
For me, running examples.py with small or `assume -s example_01a -db postgresql://assume:assume@localhost:5432/assume` works fine.

What error message do you get?

**Comment by adamsjohanna** [comment]
It stops at world.py, line 191 await self.container.as_agent_process(agent_creator=creator)
The error is AttributionError: Can't pickle local object 'enum_serializer.<locals>.__tostring__'

**Comment by nick-harder** [comment]
For me I have uninstalled and installed mango to the latest version, but it still returns the same error also on MacOS.

**Comment by maurerle** [comment]
Can one of you retest this please?

**Comment by nick-harder** [comment]
still the same error
conda activate assume-framework
(base) nick@nick-air assume % conda activate assume-framework
(assume-framework) nick@nick-air assume % /opt/anaconda3/envs/assume-framework/bin/python /Users/nick/GitHub/assume/examp
les/examples.py
Traceback (most recent call last):
  File "/Users/nick/GitHub/assume/examples/examples.py", line 68, in <module>
    load_scenario_folder(
  File "/Users/nick/GitHub/assume/assume/common/scenario_loader.py", line 534, in load_scenario_folder
    world.loop.run_until_complete(
  File "/opt/anaconda3/envs/assume-framework/lib/python3.11/site-packages/nest_asyncio.py", line 90, in run_until_complete
    return f.result()
           ^^^^^^^^^^
  File "/opt/anaconda3/envs/assume-framework/lib/python3.11/asyncio/futures.py", line 203, in result
    raise self._exception.with_traceback(self._exception_tb)
  File "/opt/anaconda3/envs/assume-framework/lib/python3.11/asyncio/tasks.py", line 267, in __step
    result = coro.send(None)
             ^^^^^^^^^^^^^^^
  File "/Users/nick/GitHub/assume/assume/common/scenario_loader.py", line 379, in load_scenario_folder_async
    await world.setup(
  File "/Users/nick/GitHub/assume/assume/world.py", line 138, in setup
    await self.setup_output_agent(simulation_id, save_frequency_hours)
  File "/Users/nick/GitHub/assume/assume/world.py", line 191, in setup_output_agent
    await self.container.as_agent_process(agent_creator=creator)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/assume-framework/lib/python3.11/site-packages/mango/container/tcp.py", line 283, in as_agent_process
    return super().as_agent_process(
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/assume-framework/lib/python3.11/site-packages/mango/container/core.py", line 839, in as_agent_process
    return self._container_process_manager.create_agent_process(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/assume-framework/lib/python3.11/site-packages/mango/container/core.py", line 462, in create_agent_process
    agent_process.start()
  File "/opt/anaconda3/envs/assume-framework/lib/python3.11/multiprocessing/process.py", line 121, in start
    self._popen = self._Popen(self)
                  ^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/assume-framework/lib/python3.11/multiprocessing/context.py", line 224, in _Popen
    return _default_context.get_context().Process._Popen(process_obj)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/assume-framework/lib/python3.11/multiprocessing/context.py", line 288, in _Popen
    return Popen(process_obj)
           ^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/assume-framework/lib/python3.11/multiprocessing/popen_spawn_posix.py", line 32, in __init__
    super().__init__(process_obj)
  File "/opt/anaconda3/envs/assume-framework/lib/python3.11/multiprocessing/popen_fork.py", line 19, in __init__
    self._launch(process_obj)
  File "/opt/anaconda3/envs/assume-framework/lib/python3.11/multiprocessing/popen_spawn_posix.py", line 47, in _launch
    reduction.dump(process_obj, fp)
  File "/opt/anaconda3/envs/assume-framework/lib/python3.11/multiprocessing/reduction.py", line 60, in dump
    ForkingPickler(file, protocol).dump(obj)
AttributeError: Can't pickle local object 'enum_serializer.<locals>.__tostring__'
(assume-framework) nick@nick-air assume %

**Comment by nick-harder** [comment]
so now it checks if it is running on linux and only then uses the different processes?

**Comment by maurerle** [comment]
Yes exactly, as other OS somehow have a serialization problem..
But the performance improvement of running the output agent in a separate process is a little less than 10% - so not game changing

File: assume/strategies/flexable.py [hunk 1]
```diff
@@ -614,23 +614,6 @@ def calculate_EOM_price_if_on(
     return bid_price_inflex
 
 
-def get_starting_costs(time, unit):
```
Comment (nick-harder) [inline assume/strategies/flexable.py:617]: why is this completely removed?

**Review by nick-harder — APPROVED** [review]

**Pull request merged**